ostelco-ios-client/Coordinators/OnboardingCoordinator.swift

```diff
@@ -377,23 +377,51 @@ extension OnboardingCoordinator: SingPassCoordinatorDelegate {
 }
 
 extension OnboardingCoordinator: MyInfoSummaryDelegate {
+    func fetchMyInfoDetails(_ controller: MyInfoSummaryViewController, code: String, completion: @escaping (MyInfoDetails) -> Void) {
+        let spinnerView = controller.showSpinner(loadingText: NSLocalizedString("Loading your data from SingPass...", comment: "Loading text after user approves SingPass"))
+        primeAPI.loadSingpassInfo(code: code)
+        .ensure {
+            controller.removeSpinner(spinnerView)
+        }
+        .done { myInfoDetails in
+            completion(myInfoDetails)
+        }
+        .catch { error in
+            ApplicationErrors.log(error)
+            controller.showGenericError(error: error) { [weak self] (_) in
+                self?.advance()
```

Comment:
You are not doing 
```
self?.localContext.selectedVerificationOption = nil
self?.localContext.myInfoCode = nil
```
before calling advance. It needs to go back to select identity verification screen